### ISSUE TITLE
Vignettes in packages that provide their own vignette engine not built

### ISSUE THREAD

**Issue by HenrikBengtsson** [r-universe-org/bugs#57]
# Issue

Package vignettes in a package that provides its own vignette engine are not built.  For an example, see my **startup** package:

* CRAN: https://cran.r-project.org/package=startup (1 vignette)
* R-universe: https://henrikbengtsson.r-universe.dev/ui#articles (no vignettes from **startup** there)

Another example is **R.rsp** which exports vignette engines but also uses them to generate it own vignette products:

* CRAN: https://cran.r-project.org/package=R.rsp (7 vignettes)
* R-universe: https://henrikbengtsson.r-universe.dev/ui#articles (no vignettes from **R.rsp** there)

# Troubleshooting

I suspect it's because the package is not installed before being built.

Interestingly, it works for rstudio/**rmarkdown** - maybe you preinstall that one since it's a commonly used package.

**Comment by jeroen** [comment]
Yes this is actually a bug in R: https://github.com/r-universe/henrikbengtsson/runs/2562839383?check_suite_focus=true

Running R CMD check on the package will bail without trying if the package is not pre-installed, which is strange because the package vignette will obviously build if the engine is provided by the package itself.
